lib/TokenManager.js

```diff
@@ -13,7 +14,60 @@ var cookieStorage = {
   }
 };
 
-function add(storage, key, token) {
+function clearRefreshTimeout(tokenMgmtRef, key) {
+  clearTimeout(tokenMgmtRef.refreshTimeouts[key]);
+  delete tokenMgmtRef.refreshTimeouts[key];
+}
+
+function clearRefreshTimeoutAll(tokenMgmtRef) {
+  var refreshTimeouts = tokenMgmtRef.refreshTimeouts;
+  for(var key in refreshTimeouts) {
+    if (!refreshTimeouts.hasOwnProperty(key)) {
+      continue;
+    }
+    clearTimeout(refreshTimeouts[key]);
+  }
+  tokenMgmtRef.refreshTimeouts = {};
+}
+
+function setRefreshTimeout(sdk, tokenMgmtRef, storage, key, token) {
+  // Wait a minimum of 0 milliseconds (the token is already expired)
+  var refreshWait = Math.max(0, (token.expiresAt * 1000) - Date.now());
+  var refreshTimeout = setTimeout(function() {
+    return refresh(sdk, tokenMgmtRef, storage, key);
+  }, refreshWait);
+
+  // Clear any existing timeout
+  clearRefreshTimeout(tokenMgmtRef, key);
+
+  // Add a new timeout
+  tokenMgmtRef.refreshTimeouts[key] = refreshTimeout;
+}
+
+function setRefreshTimeoutAll(sdk, tokenMgmtRef, storage) {
+  try {
+    var tokenStorage = storage.getTokenStorage();
+  } catch(e) {
+    // Any errors thrown on instantiation will not be caught,
+    // because there are no listeners yet
+    tokenMgmtRef.emitter.emit('error', e);
+    return;
+  }
+
+  for(var key in tokenStorage) {
+    if (!tokenStorage.hasOwnProperty(key)) {
+      continue;
+    }
+    var token = tokenStorage[key];
+    if (tokenMgmtRef.autoRefresh) {
+      setRefreshTimeout(sdk, tokenMgmtRef, storage, key, token);
+    } else {
+      tokenMgmtRef.emitter.emit('expired', key, token);
```

Comment:
Can you explain this line? And is this supposed to be what happens:
- Get tokens and store them in the TokenManager
- Refresh the page and re-initialize tokenManager with autoRefresh = false
- All my tokens emit an "expired" message